m4/mpi.m4

```diff
@@ -182,9 +198,32 @@ AS_IF([test "x$MPI_IMPL" != x],
         AC_TRY_LINK([@%:@include <mpi.h>],
                     [int np; MPI_Comm_size (MPI_COMM_WORLD, &np);],
                     [
-                       MPI_IMPL="built-in"
-                       AC_MSG_RESULT( [$CXX Compiler Supports MPI] )
-                       AC_DEFINE(HAVE_MPI, 1, [Flag indicating whether or not MPI is available])
+                    AC_TRY_RUN([@%:@include <mpi.h>
```

Comment:
I'd vote to stay away from `AC_TRY_RUN` for MPI-enabled code. Some clusters won't let you run MPI-compiled binaries on the head node, so we'd get a false positive from this test on those systems.

Reply:
you are right; this is in general considered bad style (maybe for the very same reason as you write).
However,  I understand it such that we end up here only if we didn't find the mpi.h-file, so that some grep is not accessible.

So I see two solutions: either we assume in this case that the version is >1 and leave this test away, or one tests first whether `AC_TRY_RUN` works with mpi and than checks the version.
But maybe there is a more elegant way?

Reply:
... Ok; I just found out that there is e.g. the command: mpichversion; probably there are similar ones for the other implementations and one just tries to link and if it works, we test all these commands one after the other?
This assumes that only the correct mpi-version answers; but in a well-configured environment this should be given, right?